### PR TITLE
Add hints to PyYAML dump methods

### PR DESCRIPTION
**Pull request by gaul**

### CODE REVIEW

File: third_party/2and3/yaml/__init__.pyi [hunk 1]
```diff
@@ -22,10 +22,10 @@ def safe_load_all(stream: Union[str, IO[str]]) -> Iterator[Any]: ...
 def emit(events, stream=..., Dumper=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=...): ...
 def serialize_all(nodes, stream=..., Dumper=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=..., encoding=..., explicit_start=..., explicit_end=..., version=..., tags=...): ...
 def serialize(node, stream=..., Dumper=..., **kwds): ...
-def dump_all(documents, stream=..., Dumper=..., default_style=..., default_flow_style=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=..., encoding=..., explicit_start=..., explicit_end=..., version=..., tags=...): ...
-def dump(data, stream=..., Dumper=..., **kwds): ...
-def safe_dump_all(documents, stream=..., **kwds): ...
-def safe_dump(data, stream=..., **kwds): ...
+def dump_all(documents: Sequence[Dict[str, Any]], stream=IO[str], Dumper=..., default_style=..., default_flow_style=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=..., encoding=..., explicit_start=..., explicit_end=..., version=..., tags=...) -> Any: ...
+def dump(data: Dict[str, Any], stream=IO[str], Dumper=..., **kwds) -> Any: ...
```
Comment (JelleZijlstra) [inline third_party/2and3/yaml/__init__.pyi]: It looks like documents can actually be any type: http://pyyaml.org/wiki/PyYAMLDocumentation.

File: third_party/2and3/yaml/__init__.pyi [hunk 2]
```diff
@@ -22,10 +22,10 @@ def safe_load_all(stream: Union[str, IO[str]]) -> Iterator[Any]: ...
 def emit(events, stream=..., Dumper=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=...): ...
 def serialize_all(nodes, stream=..., Dumper=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=..., encoding=..., explicit_start=..., explicit_end=..., version=..., tags=...): ...
 def serialize(node, stream=..., Dumper=..., **kwds): ...
-def dump_all(documents, stream=..., Dumper=..., default_style=..., default_flow_style=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=..., encoding=..., explicit_start=..., explicit_end=..., version=..., tags=...): ...
-def dump(data, stream=..., Dumper=..., **kwds): ...
-def safe_dump_all(documents, stream=..., **kwds): ...
-def safe_dump(data, stream=..., **kwds): ...
+def dump_all(documents: Sequence[Dict[str, Any]], stream=IO[str], Dumper=..., default_style=..., default_flow_style=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=..., encoding=..., explicit_start=..., explicit_end=..., version=..., tags=...) -> Any: ...
```
Comment (JelleZijlstra) [inline third_party/2and3/yaml/__init__.pyi]: You mean `stream: IO[str] = ...`

File: third_party/2and3/yaml/__init__.pyi (hunk 1 above)
Comment (gaul) [reply third_party/2and3/yaml/__init__.pyi]: Done.

File: third_party/2and3/yaml/__init__.pyi (hunk 2 above)
Comment (gaul) [reply third_party/2and3/yaml/__init__.pyi]: Done.

**Pull request merged**